homeassistant/components/holiday/calendar.py

```diff
@@ -0,0 +1,125 @@
+"""Holiday Calendar."""
+from __future__ import annotations
+
+from datetime import datetime
+
+from holidays import country_holidays
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_PROVINCE, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Holiday Calendar config entry."""
+    country: str = config_entry.data[CONF_COUNTRY]
+    province: str | None = config_entry.data.get(CONF_PROVINCE)
+
+    async_add_entities(
+        [
+            HolidayCalendarEntity(
+                config_entry.title,
+                country,
+                province,
+                unique_id=config_entry.entry_id,
+            )
+        ],
+        True,
+    )
+
+
+class HolidayCalendarEntity(CalendarEntity):
+    """Representation of a Holiday Calendar element."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        name: str,
+        country: str,
+        province: str | None,
+        unique_id: str,
+    ) -> None:
+        """Initialize HolidayCalendarEntity."""
+        self._country = country
+        self._province = province
+        self._location = name
+        self._event: CalendarEvent | None = None
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name=name,
+        )
+
+    def _default_language(self) -> str | None:
```

Comment:
We could probably run this once during `__init__` instead of every time.
Also can we consider that HA's language `en` is `en_US` in `holidays`? There are more differences but I guess the main one we can consider here.

Reply:
Moved to `__init__` and improved default_language.
There is a special case: For Canada, holidays has translations for `en` but not for `en_US`.